processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -158,14 +160,27 @@ public GroupByQueryQueryToolChest(
       }
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
-      final List<AggregatorFactory> aggs = Lists.newArrayList();
+      final Set<AggregatorFactory> aggs = Sets.newHashSet();
+
       for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
+        for (final AggregatorFactory transferAgg : aggregatorFactory.getRequiredColumns()) {
+          if (Iterables.any(aggs, new Predicate<AggregatorFactory>() {
+            @Override
+            public boolean apply(AggregatorFactory agg) {
+              return agg.getName().equals(transferAgg.getName()) && !agg.equals(transferAgg);
+            }
+          })) {
+            throw new IAE("Inner aggregator can currently only be referenced by a single type of outer aggregator" +
+                " for '%s'", transferAgg.getName());
+          }
+
+          aggs.add(transferAgg);
+        }
       }
 
       // We need the inner incremental index to have all the columns required by the outer query
       final GroupByQuery innerQuery = new GroupByQuery.Builder(subquery)
-          .setAggregatorSpecs(aggs)
+          .setAggregatorSpecs(Lists.newArrayList(aggs))
```

Comment:
It's a conversion of a set to a list to satisfy the method signature, not so much about copying the list.